packages/checkout/widgets-lib/src/components/Transactions/Transactions.tsx

```diff
@@ -0,0 +1,282 @@
+
+    ((rootTokenInfo.filter((result) => result.status === 'fulfilled')) as PromiseFulfilledResult<TokenInfo>[])
+      .forEach((result) => {
+        const resp = result;
+        rootData[resp.value.address!.toLowerCase()] = resp.value;
+      });
+
+    ((childTokenInfo.filter((result) => result.status === 'fulfilled')) as PromiseFulfilledResult<TokenInfo>[])
+      .forEach((result) => {
+        const resp = result;
+        childData[resp.value.address!.toLowerCase()] = resp.value;
+      });
+
+    const allTokenSymbols: string[] = [];
+    Object.values(rootData).forEach((token) => allTokenSymbols.push((token as TokenInfo).symbol.toLowerCase()));
+    Object.values(childData).forEach((token) => allTokenSymbols.push((token as TokenInfo).symbol.toLowerCase()));
+
+    cryptoFiatDispatch({
+      payload: {
+        type: CryptoFiatActions.SET_TOKEN_SYMBOLS,
+        tokenSymbols: allTokenSymbols,
+      },
+    });
+
+    return { [rootChainName]: rootData, [childChainName]: childData };
+  };
+
+  const getTransactionsDetails = async (env: Environment, address: string) => {
+    const client = new CheckoutApi({ env });
+    return client.getTransactions({ txType: TransactionType.BRIDGE, fromAddress: address });
+  };
+
+  const fetchData = async () => {
+    const address = await walletAddress();
+    if (!address) return undefined;
+
+    const localTxs = await getTransactionsDetails(checkout.config.environment, address);
+
+    const tokensWithChainSlug:{ [k:string]:string } = {};
+    localTxs.result.forEach((txn) => {
+      tokensWithChainSlug[txn.details.from_token_address] = txn.details.from_chain;
+    });
+    return { tokens: await getTokensDetails(tokensWithChainSlug), transactions: localTxs.result };
+  };
+
+  // Fetch all the data at once
+  useEffect(() => {
+    (async () => {
+      const data = await retry(fetchData, DEFAULT_TRANSACTIONS_RETRY_POLICY);
+      if (!data) return;
+
+      setKnownTokenMap(data.tokens);
+      setTxs(data.transactions);
+
+      setLoading(false);
+    })();
+  }, [walletAddress, checkout]);
+
+  return (
+    <SimpleLayout
+      testId="bridge-view"
+      header={(
+        <HeaderNavigation
+          showBack
+          title={layoutHeading}
+          onCloseButtonClick={() => sendBridgeWidgetCloseEvent(eventTarget)}
+        />
+      )}
+      footer={<FooterLogo />}
+    >
+      <Box sx={transactionsContainerStyle}>
+        <Box sx={transactionsListContainerStyle}>
+          {!provider && (
+            <EmptyStateNotConnected
+              checkout={checkout}
+              updateProvider={updateAndConnectProvider}
+            />
+          )}
+          {provider && loading
+            && (
+              <Box sx={transactionsListStyle(isPassport)}>
+                <Shimmer />
+              </Box>
+            )}
+          {provider && !loading && txs.length > 0 && knownTokenMap && (
+            <Box sx={transactionsListStyle(isPassport)}>
```

Comment:
Could we try putting this in a `useMemo` like [this](https://immutable.atlassian.net/wiki/spaces/~624b6d535f63fd0069b5765e/pages/2332917844/Styling+in+React?focusedCommentId=2349269073)? Putting it in a `useMemo` will mean that the returned value is stored in a variable in the component. This will ensure that each time the component is rendered, it does not create a new instance of the object